pinot-common/src/main/java/org/apache/pinot/common/utils/helix/LeadControllerUtils.java

```diff
@@ -58,4 +69,43 @@ public static String generatePartitionName(int partitionId) {
   public static int extractPartitionId(String partitionName) {
     return Integer.parseInt(partitionName.substring(partitionName.lastIndexOf('_') + 1));
   }
+
+  /**
+   * Checks from ZK if resource config of leadControllerResource is enabled.
+   * @param helixManager helix manager
+   */
+  public static boolean isLeadControllerResourceEnabled(HelixManager helixManager)
+      throws Exception {
+    try {
+      HelixDataAccessor helixDataAccessor = helixManager.getHelixDataAccessor();
+      PropertyKey propertyKey = helixDataAccessor.keyBuilder().resourceConfig(Helix.LEAD_CONTROLLER_RESOURCE_NAME);
+      ResourceConfig resourceConfig = helixDataAccessor.getProperty(propertyKey);
+      String resourceEnabled = resourceConfig.getSimpleConfig(Helix.LEAD_CONTROLLER_RESOURCE_ENABLED_KEY);
+      return Boolean.parseBoolean(resourceEnabled);
+    } catch (Exception e) {
+      LOGGER.warn("Could not get whether lead controller resource is enabled or not.", e);
```

Comment:
Include the full message in constructing the exception, and log it in the place where we catch it. I guess this can only be Runtime Exception. Include LEAD_CONTRROLLER_RESOURCE_NAME and LEAD_CONTROLLER_RESOURCE_ENABLED_KEY in the message.

Reply:
Removed the try catch block since any exception would be caught from the callers.